### PR TITLE
remove deprecated targets from meshery makefile

### LINKED ISSUE #6685: [CI] Cleanup old, unused make targets
#### Current Behavior
Meshery's makefile has a number of deprecated targets that need to be cleaned up.

#### Desired Behavior
Ideally, only active make targets are documented and present in the makefile.

#### Implementation
1. Update contributing.md
2. Update https://docs.meshery.io/project/contributing/contributing pages that have `make`

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by ayushthe1**
Signed-off-by: ayush_gitk <ayushsharmaa101@gmail.com>

**Notes for Reviewers**
This pr intends on removing all the deprecated targets from this [makefile](https://github.com/meshery/meshery/blob/master/Makefile)

This is a draft pr and more changes will be made here based on discussion.

This PR fixes #6685 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 06eac107d985e4c40944ee4040a5dfd1f96212a3 at: https://63adab6d6f72c13f0d95e847--meshery-docs-preview.netlify.app

File: docs/v0.5/search/index.html [hunk 1]
```diff
@@ -1770,7 +1770,7 @@ <h1></h1>
...
... by running\n\n\nmake server\n\n\nAny time changes are made to the GO code, you will have to stop the server and run the above command again.\nOnce the Meshery server is up and running, you should be able to access Meshery on your localhost on port 9081 at http://localhost:9081. One thing to note, you might NOT see the Meshery UI until the UI code is built as well.\nAfter running Meshery server, you will need to select your Cloud Provider by navigating to localhost:9081. Only then you will be able to use the Meshery UI on port 3000.\n\nBuilding Docker image\n\nTo build a Docker image of Meshery, please ensure you have Docker installed to be able to build the image. Now, run the following command to build the Docker image:\n\nmake docker\n\n\nDefine and validate errors\n\nEvery Golang-based project that belongs to Layer5 incorporates a utility to define and manage error messages for every error instance. This is internally done with several make commands, but one can explicitely validate with the help of the following make command. This checks and validates the errors that are present in the particular project.\n\nmake error\n\n\nFor more details, Error Utility\n\nSuggested Reading"
```
Comment (leecalcote) [inline docs/v0.5/search/index.html:1773]: @ayushthe1 will you undo this change to this old Meshery version's docs?

**Review by leecalcote — CHANGES_REQUESTED** [review]
Thanks, @ayushthe1! 

Will you search the entire code base for other stale references to these commands, like this one - https://github.com/meshery/meshery/blob/3a5bea9afdc1273b5dfb95af4268b8dbfbd5e8bc/CONTRIBUTING.md#L250?

**Comment by leecalcote** [comment]
@Anand-Theertha it'd be good to get your eyes on this.

**Comment by Anand-Theertha** [comment]
Hey @ayushthe1! Like I said, there are changes to be made in the readme.md file and contributing.md file. I could probably pull another PR for that, or we can connect on Slack to talk about these changes so you can modify the current PR.

**Comment by ayushthe1** [comment]
great @Anand-Theertha ,let's connect on slack and discuss this further. I will ping you today once i'm back home.

**Comment by leecalcote** [comment]
Send word when docs references have been accounted for.

**Comment by Anand-Theertha** [comment]
Yes @leecalcote, @ayushthe1 and I spoke about the changes in docs. It'll be visible after this PR is updated with the new changes.

**Comment by ayushthe1** [comment]
hey @Anand-Theertha ,i made some changes to the pr. Could you please review it.

**Comment by l5io** [comment]
🚀 Preview for commit accec28987c099be69b269bd464103ea09b29738 at: https://63b83f74da155f091f12283e--meshery-docs-preview.netlify.app

File: .github/workflows/build-ui-server-reusable-workflow.yml [hunk 2]
```diff
@@ -70,10 +70,10 @@ jobs:
           npm i
       - name: build meshery-ui
         run: |
-          make build-meshery-ui
```
Comment (leecalcote) [inline .github/workflows/build-ui-server-reusable-workflow.yml:73]: Good job on catching the changes needed in the workflows, too. 👍

**Review by Nikhil-Ladha — APPROVED** [review]

**Comment by Chadha93** [comment]
@ayushthe1  Let's discuss the update of this PR on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

**Review by leecalcote — APPROVED** [review]
Thank you, @ayushthe1!

**Pull request merged**